### PR TITLE
Adding integration test docker-compose.ci.yml and github task

### PR DESCRIPTION
**Pull request by willmeister**
## Description
Adds integration tests to CI on PR to master or merge to master.

### Fixes
- Fixes # [YAS-645](https://optimists.atlassian.net/browse/YAS-645)

## Contributing Agreement

- [x] I have read and understood the [Optimism Contributing Guide and Code of Conduct](https://github.com/ethereum-optimism/optimism-monorepo/blob/master/.github/CONTRIBUTING.md) and am following those guidelines in this pull request.

### CODE REVIEW

File: docker-compose.ci.env [hunk 1]
```diff
@@ -0,0 +1,111 @@
...
+RUN_GETH_SUBMISSION_QUEUER=1                                                                # Set to anything to run Geth Submission Queuer
+RUN_QUEUED_GETH_SUBMITTER=1                                                                 # Set to anything to run Queued Geth Submitter
+RUN_CANONICAL_CHAIN_BATCH_CREATOR=1                                                         # Set to anything to run Canonical Chain Batch Creator
+RUN_CANONICAL_CHAIN_BATCH_SUBMITTER=1                                                       # Set to anything to run Canonical Chain Batch Submitter
+RUN_STATE_COMMITMENT_CHAIN_BATCH_CREATOR=1                                                  # Set to anything to run State Commitment Chain Batch Creator
+RUN_STATE_COMMITMENT_CHAIN_BATCH_SUBMITTER=1                                                # Set to anything to run State Commitment Chain Batch Submitter
+RUN_FRAUD_DETECTOR=1                                                                        # Set to anything to run Fraud Detector
+
+
+
```
Comment (tynes) [inline docker-compose.ci.env:106]: These extra newlines could probably be deleted

File: docker-compose.ci.env [hunk 2]
```diff
@@ -0,0 +1,111 @@
...
+# Container
+STARTUP_WAIT_TIMEOUT=30                                                         # The amount of times to attempt to connect to L2 geth before failing (set to -1 if not to wait for L2 geth).
+
+# Logging
+DEBUG=info*,error*,warn*,debug*                                                 # The comma-separated logging patterns to match (common options are `error*`, `info*`, `warn*`, and `debug*`)
+
+# Postgres
+POSTGRES_HOST=postgres                                                          # (Required) The host DNS entry / IP for the postgres DB
+POSTGRES_DATABASE=rollup                                                        # (Required) The database name to connect to (should be `rollup`)
+POSTGRES_CONNECTION_POOL_SIZE                                                   # The connection pool size for postgres (defaults to 20)
```
Comment (tynes) [inline docker-compose.ci.env:39]: Just having `POSTGRES_CONNECTION_POOL_SIZE` but not set to anything will be ok?

File: docker-compose.ci.env [hunk 3]
```diff
@@ -0,0 +1,111 @@
...
+POSTGRES_USE_SSL                                                                # Set to anything to indicate that SSL should be used in the connection
```
Comment (tynes) [inline docker-compose.ci.env:40]: Will `POSTGRES_USE_SSL` fall back to non TLS? I'm not aware of any cert management happening in this setup

File: docker-compose.ci.yml [hunk 4]
```diff
@@ -0,0 +1,51 @@
...
+      - docker-compose.ci.env
+    ports:
+      - 8545:8545
+
+volumes:
+  l1-node-data:
+  l2-node-data:
+
+
+
```
Comment (tynes) [inline docker-compose.ci.yml:46]: Extra newlines can probably be deleted

**Review by tynes — APPROVED** [review]

File: docker-compose.ci.env (hunk 2 above)
Comment (willmeister) [reply docker-compose.ci.env:39]: Yep!
```
 # The connection pool size for postgres (defaults to 20)
```
☝️ it defaults to 20 when not set, and declaring the var but not providing a value means it's still not set. 

Just leaving it (and others like it) in there as documentation of what _could be_ set if a user chooses.

File: docker-compose.ci.env (hunk 3 above)
Comment (willmeister) [reply docker-compose.ci.env:40]: Yep!
```
# Set to anything to indicate that SSL should be used in the connection
```
It's not set to anything, so SSL should not be used.

**Pull request merged**